waku/v2/protocol/waku_noise/noise_utils.nim

```diff
@@ -138,6 +208,85 @@ proc hashProtocol*(protocolName: string): MDigest[256] =
 
   return hash
 
+# Commits a public key pk for randomness r as H(pk || s)
+proc commitPublicKey*(publicKey: EllipticCurveKey, r: seq[byte]): MDigest[256] =
+
+  var hashInput: seq[byte]
+  hashInput.add getBytes(publicKey)
+  hashInput.add r
+
+  # The output hash value
+  var hash: MDigest[256]
+  hash = sha256.digest(hashInput)
+
+  return hash
+
+proc genAuthcode*(hs: HandshakeState): string =
+  var output: array[1, array[8, byte]]
+  sha256.hkdf(hs.ss.h.data, [], [], output)
+  let code = cast[uint64](output[0]) mod 100_000_000
+  return $code
+
+proc initNametagsBuffer*(mntb: var MessageNametagBuffer) =
```

Comment:
That would be great if we could have an explanation of the name tag usage somewhere in RFC or in here.  Without proper description, it is hard to judge the implementation.